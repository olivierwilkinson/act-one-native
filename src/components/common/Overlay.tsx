import React from "react";
import styled from "styled-components/native";

const OverlayView = styled.View`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
`;

type Props = {
  children?: JSX.Element;
};

const Overlay: React.FunctionComponent<Props> = ({ children }) => {
  return <OverlayView pointerEvents="box-none">{children}</OverlayView>;
};

export default Overlay;
