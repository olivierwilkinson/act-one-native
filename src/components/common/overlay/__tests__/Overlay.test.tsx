import { Text } from "react-native";
import React from "react";
import { render, RenderAPI } from "react-native-testing-library";

import Overlay, { Props } from "../Overlay";

describe("Header", () => {
  let defaultProps: Props;
  let toJSON: RenderAPI["toJSON"];
  beforeEach(() => {
    defaultProps = {
      children: <Text>Overlay Content</Text>
    };

    ({ toJSON } = render(<Overlay {...defaultProps} />));
  });

  it("renders correctly", () => {
    expect(toJSON()).toMatchSnapshot();
  });
});
