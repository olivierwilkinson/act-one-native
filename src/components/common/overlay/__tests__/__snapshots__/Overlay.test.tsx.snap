// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      {
        "bottom": 0,
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": "100%",
      },
    ]
  }
>
  <Text>
    Overlay Content
  </Text>
</View>
`;
